### PR TITLE
feat(Dropdown): add checkable and checkableIndicator prop

### PR DESCRIPTION
**Pull request by mnajdova**
We have a requirement for Teams theme for adding checkmark icon in the selected dropdown item. We have several options:

1. add `selectedIndicator` icon shorthand prop in the `Dropdown`, for allowing setting checkmark on the selected drop down item:

image

2. Add `checkable` & `checkableIndicator` props on the Dropdown & DropdownItem - similar to the clearable and clearableIndicator that we already have (thanks @layershifter for the proposal)

3. Adding endMedia prop in the Dropdown Item as it is already a list item, and users can use it if they want to add some icon when the item is selected.

We need to decide on a common pattern for these things. In the Input, we have only the clearable boolean prop, so we have no way of customizing that slot. In the Dropdown together with the clearable we have the clearableIndicator. As we tend to say that every slot in the Stardust components can be customized, I believe we need to have slots for these things as well, although that they have some defaults.

### CODE REVIEW

File: docs/src/examples/components/Dropdown/Types/DropdownExample.shorthand.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ const DropdownExample = () => (
   <Dropdown
     items={inputItems}
     placeholder="Select your hero"
+    selectedIndicator="stardust-checkmark"
```
Comment (mnajdova) [inline docs/src/examples/components/Dropdown/Types/DropdownExample.shorthand.tsx]: Not sure if we should add separate example for this one..

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -285,7 +288,8 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
     search: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
     searchQuery: PropTypes.string,
     searchInput: customPropTypes.itemShorthand,
-    toggleIndicator: customPropTypes.itemShorthand,
+    selectedIndicator: customPropTypes.itemShorthandWithoutJSX,
+    toggleIndicator: customPropTypes.itemShorthandWithoutJSX,
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx]: fixed prop type

File: packages/react/src/components/Dropdown/DropdownItem.tsx [hunk 3]
```diff
@@ -43,8 +45,11 @@ export interface DropdownItemProps extends UIComponentProps<DropdownItemProps> {
    */
   onClick?: ComponentEventHandler<DropdownItemProps>
 
-  /** A dropdown item can be selected if single selection Dropdown is used. */
+  /** A dropdown item can be selected. */
   selected?: boolean
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/DropdownItem.tsx]: Added missing prop in the interface

File: packages/react/src/themes/teams/components/Dropdown/dropdownItemStyles.ts [hunk 4]
```diff
@@ -59,6 +59,10 @@ const dropdownItemStyles: ComponentSlotStylesInput<DropdownItemProps, DropdownVa
     fontSize: v.listItemContentFontSize,
     color: v.listItemContentColor,
   }),
+  selectedIndicator: ({ variables: v }) => ({
+    position: 'relative',
+    left: pxToRem(3),
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Dropdown/dropdownItemStyles.ts]: The padding on the `DropdownItem` is `11px`, but the icon should be `8px` from the right border, that's why we need styles

File: packages/react/src/components/Dropdown/DropdownItem.tsx [hunk 5]
```diff
@@ -102,6 +115,17 @@ class DropdownItem extends UIComponent<WithAsProp<DropdownItemProps>> {
             styles: styles.content,
           },
         })}
+        endMedia={
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/DropdownItem.tsx:123]: This is different then how the rest of the elements are created and where the styles are applied, but there are issues with the height if we don't provide these styles.. I am open for other ideas...

Comment (mnajdova) [reply packages/react/src/components/Dropdown/DropdownItem.tsx:123]: Ping @layershifter @miroslavstastny

**Comment by layershifter** [comment]
### Atlaskit

https://atlaskit.atlassian.com/packages/core/select
Please check `Named Exports`:

image
image

### BlueprintJS

`MultiSelect` there has similar idea:
https://blueprintjs.com/docs/#select/multi-select

image

###  AntDesign

https://ant.design/components/select/
Check multiple selection.

image

**Review by silviuaavram — APPROVED** [review]

**Comment by kolaps33** [comment]
We will need to set aria-selected on selected option:
Based on example 2 on this page:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-rearrangeable.html

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -285,7 +293,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
     search: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
     searchQuery: PropTypes.string,
     searchInput: customPropTypes.itemShorthand,
-    toggleIndicator: customPropTypes.itemShorthand,
+    toggleIndicator: customPropTypes.itemShorthandWithoutJSX,
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx:296]: fixed prop type

File: packages/react/src/themes/teams/components/Dropdown/dropdownItemStyles.ts [hunk 7]
```diff
@@ -59,6 +59,13 @@ const dropdownItemStyles: ComponentSlotStylesInput<DropdownItemProps, DropdownVa
     fontSize: v.listItemContentFontSize,
     color: v.listItemContentColor,
   }),
+  checkableIndicator: ({ variables: v }) => ({
+    position: 'relative',
+    left: pxToRem(3),
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Dropdown/dropdownItemStyles.ts:64]: The padding on the `DropdownItem` is `11px`, but the icon should be `8px` from the right border, that's why we need styles

**Review by kolaps33 — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -29,6 +29,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Add `onActiveIndexChange` prop to `Tree` component @lucivpav ([#1728](https://github.com/stardust-ui/react/pull/1728))
 - Add `horizontal` prop for `List` component @mnajdova ([#1721](https://github.com/stardust-ui/react/pull/1721))
 - Export `call-blocked` icon to Teams theme @francescopalmiotto ([#1736](https://github.com/stardust-ui/react/pull/1736))
+- add `checkable` and `checkableIndicator` to the `Dropdown` and `DropdownItem` components @mnajdova ([#1738](https://github.com/stardust-ui/react/pull/1738))
```
Comment (layershifter) [inline CHANGELOG.md]:
```suggestion
- Add `checkable` and `checkableIndicator` to the `Dropdown` and `DropdownItem` components @mnajdova ([#1738](https://github.com/stardust-ui/react/pull/1738))
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**